front/lib/resources/storage/models/vaults.ts

```diff
@@ -54,7 +54,10 @@ VaultModel.init(
   {
     modelName: "vaults",
     sequelize: frontSequelize,
-    indexes: [{ unique: false, fields: ["workspaceId", "kind"] }],
+    indexes: [
+      { unique: true, fields: ["workspaceId", "name"] },
```

Comment:
If we're relying on the DB index to ensure unique vault names per workspace, it's important to handle errors appropriately, as the current approach will simply throw an exception.